frontend/src/services/QueryServiceImpl.ts

```diff
@@ -0,0 +1,61 @@
+import { injectable } from 'inversify';
+import 'reflect-metadata';
+import { LimitQuery } from '../entities/queries/LimitQuery';
+import { QueryResult } from '../entities/queries/QueryResult';
+import CancellationError from '../utils/CancellationError';
+import { CancellationToken } from '../utils/CancellationToken';
+import nop from '../utils/nop';
+import QueryService from './QueryService';
+
+@injectable()
+export default class QueryServiceImpl extends QueryService {
+  private readonly baseUri = 'http://localhost:8080'; // TODO: Make me configurable
+
+  public queryAll(
+    query?: LimitQuery,
+    cancellation?: CancellationToken
+  ): Promise<QueryResult> {
+    const url = `${this.baseUri}/queryAll`;
+
+    return new Promise<QueryResult>((resolve, reject) => {
```

Comment:
Can we put all that "http-stuff" with cancel events etc. in a new class? We surely need this functionality in other methods as well.
But its good enough ATM.

Reply:
Had this in mind, but as this is just needed in this single place currently did not invest the effort to actually factor it out yet.